### PR TITLE
feat(gatsby-parcel-config): create parcel config package

### PR DESCRIPTION
**Pull request by marvinjude**
## Description

The Typescript POC uses `@parcel/config-default` as `defaultConfig` which is quite heavy  as it provides support for things we don't really care about e.g. Vue, css related transformers, ...

This PR create a minimal [parcel config](https://parceljs.org/plugin-system/configuration/) package.

[sc-45870]

### CODE REVIEW

File: packages/gatsby-parcel-config/index.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (LekoArts) [inline packages/gatsby-parcel-config/index.json:1]: You'll want to check the `transformers` and remove more that won't be relevant for Gatsby code, e.g. pug or coffeescript

File: packages/gatsby-parcel-config/package.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "gatsby-parcel-config",
+  "main": "index.json",
+  "version": "1.0.0",
```
Comment (LekoArts) [inline packages/gatsby-parcel-config/package.json:4]:
```suggestion
  "version": "0.0.1-next.0",
```

File: packages/gatsby/package.json [hunk 3]
```diff
@@ -86,6 +85,7 @@
     "gatsby-legacy-polyfills": "^2.8.0-next.0",
     "gatsby-link": "^4.8.0-next.0",
     "gatsby-page-utils": "^2.8.0-next.0",
+    "gatsby-parcel-config": "^1.0.0",
```
Comment (LekoArts) [inline packages/gatsby/package.json:88]:
```suggestion
    "gatsby-parcel-config": "^0.0.1-next.0",
```

Comment (marvinjude) [reply packages/gatsby/package.json:88]: Thanks!

**Comment by marvinjude** [comment]
Seems good now

**Comment by marvinjude** [comment]
@LekoArts For watching changes, how do we typically handle "no-code" packages? We probably need to put `index.json` in some `src` directory and copy over to root or so when file changes.

**Review by LekoArts — APPROVED** [review]

**Pull request merged**